Dockerfile

```diff
@@ -0,0 +1,5 @@
+FROM golang:1.11.3-stretch
+COPY . /go/src/kind
+WORKDIR /go/src/kind
+RUN go get .
+RUN CGO_ENABLED=0 GOOS=linux go install -a -ldflags '-extldflags "-static"' .
```

Comment:
if we are adding a Docker file, we might as well `go get sigs.k8s.io/kind` from inside the container and allow the user to specify a kind tag.
also add docs on how to `docker run` and mount a volume where the binary will be built.

with this current Docker file it is expected from the user to have already downloaded the project source code "somehow".

Reply:
Besides those suggestions, this assumes Linux, while the project supports Mac and is working on Windows. Ideally we would allow cross compiling.

I'd also rather not add more files to the root of the project, this should go in something like hack/build with instructions for using it.

Reply:
@zoobab one detail that's missing is how what you are suggesting will help make getting started with kind easier than simply downloading a binary for a given architecture?

Reply:
I will add a build.sh script and change the ENTRYPOINT to copy the binary. I don't know how those released binaries are produced/compiled, but my experience with Github releases is that it comes from the dev laptop, or travis-ci integration at best. But a simple Dockerfile outputting a binary, with multiple devs obtaining the same md5sum at the end should be a better release process IMHO.

Reply:
I will add the same entrypoint and build.sh as I did here: https://github.com/zoobab/terraform-provider-infoblox/blob/Dockerfile/Dockerfile and https://github.com/zoobab/terraform-provider-infoblox/blob/Dockerfile/build.sh .